### PR TITLE
Car details and create reservation front end

### LINKED ISSUE #5: [2pt] Design and Implement Car Details Component with API Interaction
Create a component to showcase car information and booking potential. Implement API interaction for fetching specific car data.

### LINKED ISSUE #6: [2pt] Design and Implement Reservation Form with Input Validation and API Interaction
Create a form for users to input booking details (dates, user autofill). Implement user-friendly forms with data integrity checks. Implement API interaction for posting reservation data.

### PR DESCRIPTION
**Pull request by tomasesquivelgc**
# In this branch I:

## CarDetails:
- Give styles to the carDetails page
- Display information in a table (like template)
- Add functionality to template elements (Discover more)
- Add a chromatic circle PNG to match template (no functionality)
- Add a "Reserve now button" that redirects to the AddReservepage
- Pass down the carID as a query parameter

## AddReserve
- Give styles to the page
- Add a form with a working submit button
- Ensure that the reservations are being created in the database
- Use the carID query parameter to initialize the car selector (when you click on reserve button from CarDetails, the initial state of the car selector will be the one from where you came.)
- Add a background image with green gradient to match template.

CLOSES #5
CLOSES #6

### CODE REVIEW

File: src/pages/CarDetails.jsx [hunk 1]
```diff
@@ -33,75 +54,59 @@ const CarDetails = () => {
...
                 <img
-                  className="w-full h-auto"
+                  className="w-full h-auto p-4"
                   src={currentCar?.image_url}
-                  alt="Vespa Scooter"
+                  alt={currentCar?.name}
                 />
+                <Link to="/">
+                  <img src={triangle} className="bg-light-green w-16 transform -rotate-90 rounded-b-xl" alt="Home" />
+                </Link>
```
Comment (Estete9) [inline src/pages/CarDetails.jsx]: Great job with this component, it is looking great 👍 
I would only make one suggestion about this back button. As it is now, it is below the car image, and depending on the size of the image the button changes position (like in the screenshot below). 
I can think of three options
- Make the car image size the same for all cars, 
- Maybe it's a good idea to put the button on top of the car image 
- Or another option is to have it fixed in a position (and over the image)

image

File: src/pages/CarDetails.jsx [hunk 2]
```diff
@@ -33,75 +54,59 @@ const CarDetails = () => {
...
+                  <p className="font-bold">{currentCar?.car_detail.horsepower}</p>
+                  <p>&nbsp;Horse power</p>
+                </div>
+                <div className="w-full p-3 mx-4">
+                  <button
+                    type="button"
+                    className="cursor-pointer text-sm font-bold hover:underline"
+                    onClick={() => handleShowMoreDetails()}
+                  >
+                    {showMoreDetails ? 'CLOSE DETAILS △' : 'DISCOVER MORE DETAILS ▷'}
```
Comment (Estete9) [inline src/pages/CarDetails.jsx:94]: [OPTIONAL]
This logic is working amazing man! great job!
I could suggest that we use the same color as the other buttons so it is easier for the user to know that this is a button. 
The color could be to the whole button's text or only the triangle symbol

**Review by Estete9 — CHANGES_REQUESTED** [review]
Hi @tomasesquivelgc 
I'll review this PR. ✋

Good job so far it looks amazing Tom!
There are some recommendations I would like you to take into account.



Highlights
 You have written a clean code ✅
 The distribution, padding, and margins are perfect. ✅
 Good PR title/description ✅
Required Changes ♻️
 Good job adding the elements in the list in the main section, but we can improve it a little bit more together. 💪 
 
Check the comments under the review.

Cheers and Happy coding!👏👏👏

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @Estete9 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/Estete9">
   @Estete9
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/dev-esteban-palacios/">
   linkedin-esteban-palacios-5030a772
</a>

</td>
<td>

<a href="https://twitter.com/NaughTban">
   @NaughTban
</a>

</td>
</tr>

File: src/pages/CarDetails.jsx (hunk 2 above)
Comment (tomasesquivelgc) [reply src/pages/CarDetails.jsx:94]: I like that idea, makes more clear the user interaction. I'll keep it in mind for future features, for now I'll decide to stick to the original template and move forward with other things. Thank you for your comment 🙏

**Review by Estete9 — APPROVED** [review]
## Hi @tomasesquivelgc , 

Your feature is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 




## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Great use of modules ✅ 
- Linters are green ✅ 
- Meaningful **Github** commits ✅ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @Estete9 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/Estete9">
   @Estete9
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/dev-esteban-palacios/">
   linkedin-esteban-palacios-5030a772
</a>

</td>
<td>

<a href="https://twitter.com/NaughTban">
   @NaughTban
</a>

</td>
</tr>
</table>

**Pull request merged**